code/controllers/subsystem/input.dm

```diff
@@ -1,128 +1,32 @@
-			),
-		"old_default" = list(
-			"Tab" = "\".winset \\\"mainwindow.macro=old_hotkeys map.focus=true input.background-color=[COLOR_INPUT_DISABLED]\\\"\"",
-			"Ctrl+T" = ".say",
-			"Ctrl+O" = "ooc",
-			),
-		"old_hotkeys" = list(
-			"Tab" = "\".winset \\\"mainwindow.macro=old_default input.focus=true input.background-color=[COLOR_INPUT_ENABLED]\\\"\"",
-			"O" = "ooc",
-			"T" = ".say",
-			"M" = ".me",
-			"Back" = "\".winset \\\"input.focus=true input.text=\\\"\\\"\\\"\"", // This makes it so backspace can remove default inputs
-			"Any" = "\"KeyDown \[\[*\]\]\"",
-			"Any+UP" = "\"KeyUp \[\[*\]\]\"",
-			),
-		)
-
-	// Because i'm lazy and don't want to type all these out twice
-	var/list/old_default = default_macro_sets["old_default"]
-
-	var/list/static/oldmode_keys = list(
-		"North", "East", "South", "West",
-		"Northeast", "Southeast", "Northwest", "Southwest",
-		"Insert", "Delete", "Ctrl", "Alt",
-		"F1", "F2", "F3", "F4", "F5", "F6", "F7", "F8", "F9", "F10", "F11", "F12",
-		)
-
-	for(var/i in 1 to oldmode_keys.len)
-		var/key = oldmode_keys[i]
-		old_default[key] = "\"KeyDown [key]\""
-		old_default["[key]+UP"] = "\"KeyUp [key]\""
-
-	var/list/static/oldmode_ctrl_override_keys = list(
-		"W" = "W", "A" = "A", "S" = "S", "D" = "D", // movement
-		"1" = "1", "2" = "2", "3" = "3", "4" = "4", // intent
-		"B" = "B", // resist
-		"E" = "E", // quick equip
-		"F" = "F", // intent left
-		"G" = "G", // intent right
-		"H" = "H", // stop pulling
-		"Q" = "Q", // drop
-		"R" = "R", // throw
-		"X" = "X", // switch hands
-		"Y" = "Y", // activate item
-		"Z" = "Z", // activate item
-		)
-
-	for(var/i in 1 to oldmode_ctrl_override_keys.len)
-		var/key = oldmode_ctrl_override_keys[i]
-		var/override = oldmode_ctrl_override_keys[key]
-		old_default["Ctrl+[key]"] = "\"KeyDown [override]\""
-		old_default["Ctrl+[key]+UP"] = "\"KeyUp [override]\""
-
-	macro_sets = default_macro_sets
+	/// List of clients whose input to process in loop.
+	var/list/processing = list()
 
-// For initially setting up or resetting to default the movement keys
-/datum/controller/subsystem/input/proc/setup_default_movement_keys()
-	var/static/list/default_movement_keys = list(
-		"W" = NORTH, "A" = WEST, "S" = SOUTH, "D" = EAST,				// WASD
-		"North" = NORTH, "West" = WEST, "South" = SOUTH, "East" = EAST,	// Arrow keys & Numpad
-		)
-	var/static/list/azerty_movement_keys = list(
-		"Z" = NORTH, "Q" = WEST, "S" = SOUTH, "D" = EAST,				// WASD
-		"North" = NORTH, "West" = WEST, "South" = SOUTH, "East" = EAST,	// Arrow keys & Numpad
-	)
-	movement_keys = default_movement_keys.Copy()
-	alt_movement_keys = azerty_movement_keys.Copy()
+/datum/controller/subsystem/input/fire(resumed = FALSE)
+	var/list/to_cull
+	for(var/c in processing)
+		var/client/C = c
+		if(processing[C] + AUTO_CULL_TIME < world.time)
+			if(!length(C.input_data.keys_held))
+				LAZYADD(to_cull, C)
+			else
+				continue // they fell asleep on their keyboard or w/e, let them
```

Comment:
Is this meant to skip the clients key loop if they are holding down keys? I am worried about cases such as people using spacepods where they may hold a direction key for more than 60 seconds and then wonder why it stops.